File: pinyougou-manager-web/src/main/java/com/pinyougou/controller/BrandController.java
package com.pinyougou.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll.do")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }


    @RequestMapping("/findPage.do")
    public PageResult findPage(int pageNum, int pageSize,@RequestBody TbBrand brand){
        return brandService.findPage(brand,pageNum,pageSize);
    }

    @RequestMapping("/save.do")
    public Result save(@RequestBody TbBrand brand){
        try {
            brandService.save(brand);
            return new Result(true,"保存成功");
        } catch (Exception e) {
            return new Result(false,"保存失败");
        }
    }

    @RequestMapping("/update.do")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"保存成功");
        } catch (Exception e) {
            return new Result(false,"保存失败");
        }
    }


    @RequestMapping("/findOne.do")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/delete.do")
    public void delete(Long[] ids){
        for (Long id : ids) {
            brandService.delete(id);
        }
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("brandController",function ($scope,brandService,$controller) {//定义控制器

    $controller("baseController",{$scope:$scope})

    $scope.findAll = function(){//查询所有不分页
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    }


    $scope.search = function(pageNum,pageSize){
        //发送ajax 传递分页参数
        brandService.search($scope.searchEntity,pageNum,pageSize).success(
            //返回值json数据
            function (response) {
                $scope.list = response.rows;//把分页结果放入变量展示
                $scope.paginationConf.totalItems = response.total;//将总条数赋值给分页插件
            }
        )
    }

    //修改之前的回显方法
    $scope.findOne = function(id){
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response
            }
        )
    }

    //保存方法，判断有id就调用后台修改方法，没有id就调用保存方法
    $scope.save = function () {
        var object = null;
        if ($scope.entity.id != null) {
            object = brandService.update($scope.entity)
        }else{
            object = brandService.add($scope.entity)
        }
        object.success(
            function (response) {
                if (response.result){
                    $scope.reloadList();
                } else {
                    alert(response.msg);
                }
            }
        )
    }

    //点击删除调用的删除方法
    $scope.delete = function () {
        brandService.delete($scope.selectIds).success(
            function () {
                $scope.reloadList();
            }
        )
    }
})